chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1891,11 +1893,50 @@ def run_stratagem(self, mdts, fs_id, stratagem_data):
                 "keep_failed": False,
                 "actions": ["stratagem.purge"],
             }
+            action = "purge"
             task = Task.objects.create(**task_data)
             task_list.append(task)
 
             run_stratagem_list.append({"class_name": "CreateTaskJob", "args": {"task": task}})
 
+        if stratagem_data.get("filesync"):
+            task_data = {
+                "filesystem": filesystem,
+                "name": "{}-filesync-filesync".format(unique_id),
+                "start": django.utils.timezone.now(),
+                "state": "created",
+                "keep_failed": False,
+                "actions": ["stratagem.filesync"],
+                "args": {
+                    "remote": stratagem_data.get("remote"),
+                    "expression": stratagem_data.get("expression"),
+                    "action": stratagem_data.get("action"),
+                },
+            }
+            action = "filesync"
+            task = Task.objects.create(**task_data)
+
+            run_stratagem_list.append({"class_name": "CreateTaskJob", "args": {"task": task}})
```

Comment:
If this gets called with both filesync and cloudsync set, it will leak a filesync Task

Reply:
Done